server/controllers/userControllers/rentals.js

```diff
@@ -0,0 +1,71 @@
+const Rental = require("../../models/rental");
+const Car = require("../../models/car");
+
+exports.postAddRental = async (req, res) => {
+  const { car, startDate, endDate, user, paid } = req.body;
+
+  try {
+    const overlappingRentals = await Rental.find({
+      car: car,
+      startDate: { $lte: endDate }, 
+      endDate: { $gte: startDate }
+    });
+
+    if (overlappingRentals.length > 0) {
+      return res.status(400).json({
+        message: "Car is already rented for the requested dates."
+      });
+    }
```

Comment:
What if we have more than one car of the same type ? If we treat each car as a different document in database even if it is the same model - this is ok. But other approach is to store number of available and all cars of given model for each car document.